src/Tools/dotnet-monitor/Exceptions/ExceptionsStore.cs

```diff
@@ -104,24 +104,74 @@ private async Task ProcessEntriesAsync(CancellationToken token)
                         moduleName = NameFormatter.GetModuleName(entry.Cache.NameCache, exceptionClassData.ModuleId);
                     }
 
+                    CallStackResult callStackResult = GenerateCallStackResult(entry.StackFrameIds, entry.Cache, entry.ThreadId);
+
                     lock (_instances)
                     {
-                        _instances.Add(new ExceptionInstance(exceptionTypeName, moduleName, entry.Message, entry.Timestamp));
+                        _instances.Add(new ExceptionInstance(exceptionTypeName, moduleName, entry.Message, entry.Timestamp, callStackResult));
                     }
                 }
 
                 shouldReadEntry = await _channel.Reader.WaitToReadAsync(token);
             }
         }
 
+        internal static CallStackResult GenerateCallStackResult(ulong[] stackFrameIds, IExceptionsNameCache cache, int threadId)
+        {
+            CallStackResult callStackResult = new CallStackResult();
+            CallStack callStack = new();
+            callStack.ThreadId = (uint)threadId;
+
+            foreach (var stackFrameId in stackFrameIds)
+            {
+                if (cache.TryGetStackFrameIds(stackFrameId, out ulong methodId, out int ilOffset))
+                {
+                    if (cache.TryGetFunctionId(methodId, out FunctionData functionData))
+                    {
+                        callStackResult.NameCache.FunctionData.TryAdd(methodId, functionData);
+
+                        if (cache.TryGetClassId(functionData.ParentClass, out ClassData classData))
+                        {
+                            callStackResult.NameCache.ClassData.TryAdd(functionData.ParentClass, classData);
+
+                            ModuleScopedToken moduleScopedToken = new(functionData.ModuleId, classData.Token);
+
+                            if (cache.TryGetToken(moduleScopedToken, out TokenData tokenData))
+                            {
+                                callStackResult.NameCache.TokenData.TryAdd(moduleScopedToken, tokenData);
+                            }
+                        }
+                        if (cache.TryGetModuleId(functionData.ModuleId, out ModuleData moduleData))
+                        {
+                            callStackResult.NameCache.ModuleData.TryAdd(functionData.ModuleId, moduleData);
+                        }
+                    }
+
+                    CallStackFrame frame = new()
+                    {
+                        FunctionId = methodId,
+                        Offset = (ulong)ilOffset // is this safe?
```

Comment:
IIRC iloffset as reported by the stacks api is an absolute memory location (hence the ulong). I believe you are reporting a relative to function iloffset here correct?

Reply:
Correct, this is the relative managed ILOffset within the method body; I don't think we have access to the native IL Offset.